pkg/workflows/management/core_components.go

```diff
@@ -92,10 +92,13 @@ func runUpgradeCoreComponents(ctx context.Context, commandContext *task.CommandC
 		commandContext.SetError(err)
 		return err
 	}
-	commandContext.ClusterSpec.Cluster.SetManagementComponentsVersion(commandContext.ClusterSpec.EKSARelease.Spec.Version)
+
+	// ManagedFields must be nil to perform the server-side apply or a BadRequest error is thrown.
+	commandContext.CurrentClusterSpec.Cluster.ManagedFields = nil
+	commandContext.CurrentClusterSpec.Cluster.SetManagementComponentsVersion(commandContext.CurrentClusterSpec.EKSARelease.Spec.Version)
 	if err := client.ApplyServerSide(ctx,
-		constants.EKSAFieldManager,
-		commandContext.ClusterSpec.Cluster,
+		constants.EKSACLIFieldManager,
+		commandContext.CurrentClusterSpec.Cluster,
 		kubernetes.ApplyServerSideOptions{ForceOwnership: true},
```

Comment:
Updated this. It should have been using the current cluster spec, we don't want to apply any other cluster updates here.